lib/rules/declaration-block-no-duplicate-properties/index.js

```diff
@@ -88,97 +88,68 @@ const rule = (primary, secondaryOptions, context) => {
-									decl.remove();
-								} else {
-									removePreviousDuplicate(decls, lowerProp);
-								}
-
-								return;
-							}
-
-							report({
-								message: messages.rejected(prop),
-								node: decl,
-								result,
-								ruleName,
-								word: prop,
-							});
-
-							return;
-						}
-
-						if (ignorePrefixlessSameValues) {
-							// fails if values of consecutive, unprefixed duplicates are equal
-							if (vendor.unprefixed(value) !== vendor.unprefixed(duplicateValue)) {
-								if (context.fix) {
-									if (!important && duplicateImportant) {
-										decl.remove();
-									} else {
-										removePreviousDuplicate(decls, lowerProp);
-									}
-
-									return;
-								}
-
-								report({
-									message: messages.rejected(prop),
-									node: decl,
-									result,
-									ruleName,
-									word: prop,
-								});
-
-								return;
-							}
-						}
+				if (indexDuplicate === -1) {
+					decls.push(decl);
+				}
 
-						// fails if values of consecutive duplicates are equal
-						if (value === duplicateValue) {
-							if (context.fix) {
-								removePreviousDuplicate(decls, lowerProp);
+				const duplicateDecl = decls[indexDuplicate];
 
-								return;
-							}
+				if (!duplicateDecl) {
+					return;
+				}
 
-							report({
+				const duplicateValue = duplicateDecl.value || '';
+				const duplicateImportant = duplicateDecl.important || false;
+				const duplicateIsMoreImportant = !important && duplicateImportant;
+				const duplicatesAreConsecutive = indexDuplicate === decls.length - 1;
+				const unprefixedDuplicatesAreEqual =
+					vendor.unprefixed(value) === vendor.unprefixed(duplicateValue);
+
+				if (ignoreDiffValues || ignorePrefixlessSameValues) {
+					if (
+						!duplicatesAreConsecutive ||
+						(ignorePrefixlessSameValues && !unprefixedDuplicatesAreEqual)
+					) {
+						if (!context.fix) {
+							return report({
 								message: messages.rejected(prop),
 								node: decl,
 								result,
 								ruleName,
 								word: prop,
 							});
+						}
 
-							return;
+						if (duplicateIsMoreImportant) {
+							return decl.remove();
 						}
 
-						return;
+						return duplicateDecl.remove();
 					}
 
-					if (ignoreDuplicates && indexDuplicate === decls.length - 1) {
+					if (value !== duplicateValue) {
 						return;
 					}
 
 					if (context.fix) {
-						if (!important && duplicateImportant) {
-							decl.remove();
-						} else {
-							removePreviousDuplicate(decls, lowerProp);
-						}
-
-						return;
+						return duplicateDecl.remove();
 					}
 
-					report({
+					return report({
+						message: messages.rejected(prop),
+						node: decl,
+						result,
+						ruleName,
+						word: prop,
+					});
+				}
+
+				if (ignoreDuplicates && duplicatesAreConsecutive) {
+					return;
+				}
+
+				if (!context.fix) {
+					return report({
 						message: messages.rejected(prop),
 						node: decl,
 						result,
```

Comment:
Is
```
  ruleName,
  word: prop,
```
missing?

Reply:
Why do you think it's missing? It's right there

Reply:
Not on github. Maybe you forgot to push?